cavern/src/main/java/org/opencadc/cavern/files/GetAction.java

```diff
@@ -116,8 +120,18 @@ public void doAction() throws Exception {
                 return;
             }
             
-            // TODO: get node so we can put appropriate properties into HTTP headers
+            // set HTTP headers
+            Path rootPath = Paths.get(getRoot());
+            Node node = NodeUtil.get(rootPath, nodeURI);
```

Comment:
Are you sure this works correctly when there are links in the path? I had to change NodeUtil.get to work with PathResolver and that meant returning a LinkNode when I ran into one... line 439 of NodeUtil. basically, I think the break on line 446 means return null - maybe that needs to change to resolve the link to a container and continue? (eg call pathToNode again but without using LinkOption.NOFOLLOW_LINKS (on line 493)... I'm not sure and the NodeUtilTest isn't thorough enough

Reply:
I'll have to think about this more.  Yes, more tests are probably needed.